### ISSUE TITLE
no-descending-specificity false positive

### ISSUE THREAD

**Issue by revyh** [stylelint/stylelint#969]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

`no-descending-specificity` is reporting false positives when run from within Atom's linter-stylelint. Originally reported under AtomLinter/linter-stylelint#133 by @smutnyleszek .

> Which rule, if any, is this issue related to?

`no-descending-specificity`

> What CSS is needed to reproduce this issue?

```css
a {}
b a {}
```

> What stylelint configuration is needed to reproduce this issue?

```yaml
---
  rules:
    no-descending-specificity: [true, {severity: error}]
```

Source files [blah.zip](https://github.com/stylelint/stylelint/files/189051/blah.zip)

You may be needed to save `blah.css` couple of times to see the bug.

> Which version of stylelint are you using?

`5.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Via the Node API within `linter-stylelint`.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No false positive.

> What actually happened (e.g. what warnings or errors you are getting)?

![_2016-03-25_11-11-33](https://cloud.githubusercontent.com/assets/668565/14037779/c67d9448-f282-11e5-80f0-a029d92a1ffd.png)

As you can see, Atom's `linter-stylelint` reports an error while CLI invoked stylelint (globally installed) reports no errors.

> My guess

[no-descending-specificity rule](https://github.com/stylelint/stylelint/blob/5.2.0/src/rules/no-descending-specificity/index.js) uses [`getContext` function from `cssNodeContextLookup` util](https://github.com/stylelint/stylelint/blob/5.2.0/src/utils/cssNodeContextLookup.js#L15). getContest uses [`contextMap` const](https://github.com/stylelint/stylelint/blob/5.2.0/src/utils/cssNodeContextLookup.js#L12) from it's own closure. So, because styelint instance in `linter-stylelint` is not going to die after each Node API `lint` call, `contextMap` accumulates contexts from all previous styelint's runs.

**Comment by davidtheclark** [comment]
Thanks for the figuring it out! Looks like exactly the same issue as https://github.com/stylelint/stylelint/issues/968. Will fix.

**Comment by revyh** [comment]
Thanks for the quick fix!
